src/main/java/dan200/computercraft/core/filesystem/FileSystem.java

```diff
@@ -697,11 +729,7 @@ public void flush() throws IOException
                     throw new UnsupportedOperationException();
                 }
             };
-            synchronized( m_openFiles )
-            {
-                m_openFiles.add( file );
-            }
-            return file;
+            return (IMountedFileNormal) openFile( file, reader );
```

Comment:
I generally don't like casts unless you need them. I guess you could make `openFile` generic:

```java
private synchronized <T extends IMountedFile> T openFile( T file, Closeable handle ) throws FileSystemException
```

Though I'm not sure that's any nicer 😄.